src/ast/passes/builtins.cpp

```diff
@@ -106,6 +107,29 @@ std::optional<Expression> Builtins::visit(Call &call)
         return ast_.make_node<Integer>(str->loc, signal_num);
       }
     }
+  } else if (call.func == "__builtin_kfunc_exist") {
+    if (call.vargs.size() != 1) {
+      call.addError() << "__builtin_kfunc_exist expects 1 argument";
+    } else {
+      String *kfunc = call.vargs.at(0).as<String>();
+      return ast_.make_node<Boolean>(
+          kfunc->loc, bpftrace_.feature_->has_kfunc(kfunc->value));
+    }
+  } else if (call.func == "__builtin_kfunc_allowed") {
+    if (call.vargs.size() != 1) {
+      call.addError() << "__builtin_kfunc_allowed expects 1 argument";
+    } else {
+      String *kfunc = call.vargs.at(0).as<String>();
+      auto *probe = dynamic_cast<Probe *>(top_level_node_);
+      if (!probe) {
+        return std::nullopt;
```

Comment:
Nit: we should use `LOG(BUG)` here if we can't get the probe from the top_level_node_